### PR TITLE
Feature/footer with toggle

### PR DESCRIPTION
**Pull request by CedricPoon-SeekAsia**
Serves as deployment of FooterHK 2.1.
Details on https://trello.com/c/qw8I7RrR/54-footer-21

BREAKING CHANGE:

Enables FooterHK to support toggling expansion

### CODE REVIEW

File: jobsDB/Footer/FooterHK/FooterHK.less [hunk 1]
```diff
@@ -19,22 +21,64 @@
...
   padding-top: @wrapper-up-padding-top;
-  padding-bottom: @wrapper-low-vertical-padding;
+
+  -webkit-transition: all 0.3s ease-in-out;
+  -moz-transition: all 0.3s ease-in-out;
+  -ms-transition: all 0.3s ease-in-out;
+  -o-transition: all 0.3s ease-in-out;
+  transition: all 0.3s ease-in-out;
+
+  &.hidden {
```
Comment (mengtzu) [inline jobsDB/Footer/FooterHK/FooterHK.less]: Just be careful with this, you end up with 1px of the thing visible, which can look like a double thick border or something and be really confusing.

File: jobsDB/Footer/FooterHK/FooterHK.less [hunk 2]
```diff
@@ -19,22 +21,64 @@
...
+  @media @mobile {
+    flex-direction: column;
+    line-height: @grid-base * 4;
   }
 }
 
 .copyright {
   text-align: center;
-  padding-top: @copyright-padding-top;
+  @media @mobile {
```
Comment (mengtzu) [inline jobsDB/Footer/FooterHK/FooterHK.less]: Can you refactor this to be mobile first?  `@mobile` caused us some problems when we changed the breakpoints recently.

File: jobsDB/Footer/FooterHK/components/FooterLinks/FooterLinks.less [hunk 3]
```diff
@@ -10,6 +10,11 @@
   align-items: center;
   justify-content: center;
   line-height: @list-line-height;
+  @media @mobile {
```
Comment (mengtzu) [inline jobsDB/Footer/FooterHK/components/FooterLinks/FooterLinks.less]: Again (and for all other `@mobile`) please refactor to mobile first.

File: react/DropdownList/DropdownList.js [hunk 4]
```diff
@@ -33,11 +35,13 @@ export default class DropdownList extends Component {
           [styles.lowerLevel]: lowerLevel,
           [className]: className
         })} >
-        <div className={styles.toggler} onClick={this.toggleDropdown.bind(this)} >
+        <div
+          className={classnames({ [styles.toggler]: true, [styles.position]: iconPosition === 'left' })}
```
Comment (mengtzu) [inline react/DropdownList/DropdownList.js]: Use the constants defined rather than a string for this comparison?

**Comment by CedricPoon-SeekAsia** [comment]
@mengtzu @swetharajendra05 All the suggestions are considered and done code refinement.

**Comment by mengtzu** [comment]
This seems like it isn't hooked up to the prop to offset the action tray height.  I think we'll have an issue if this is used alongside the header without that 🤔

**Comment by swetharajendra05** [comment]
@mengtzu : Sorry for the delay, I have added some props to retain the expanded version of the footer and toggled version. Can you please help to review once.

**Comment by mengtzu** [comment]
Hi @swetharajendra05 - I'm still a little bit confused about the `toggleVersion` prop and what it's supposed to do.  I can see it renders a chevron which changes the state (and thereby the orientation of the chevron) but in the preview (seek-asia-style-guide--55dffceccc004489cced773e0a8fb300c765e2ec.surge.sh) it doesn't seem to do much else.

I'm also reasonably sure this is going to run into trouble with the header action tray on small screens?  I can't see the offset for it here.

Let me know if you want to pair on this PR to get it through faster :)

**Comment by swetharajendra05** [comment]
@mengtzu : not sure why the "toggleVersion" prop  it is not being rendered on the preview env, but for your reference added the screen shot:
toggled version(prop to enable it: toggleVersion): is the one with the chevron icon, which will toggle the expanded and thin version of footer and this would be used on the results page and split view mode. 
screen shot 2018-10-25 at 8 04 56 am

screen shot 2018-10-25 at 11 22 17 am


isExpandedVersion: is just the expanded version of the footer which would be used on the Homepage and it would not have any chevron icon tagged to it and it will also not have any sliding feature.
screen shot 2018-10-25 at 8 04 28 am

please do let me know if this is unclear

**Comment by patrickliu-sa** [comment]
Can we make the props more explicit about what each of them enabling? e.g.
- prop `Legacy` for all the old footers, with copyright info floated right
- the `toggleVersion` becomes the default (no prop required)
- prop `Full` for expanded footer without chervon to collapse
- prop `Expandable` with chervon icon, when checked `Full` will be true

**Comment by swetharajendra05** [comment]
@patrickliu-sa : I have discussed with Jye and renamed the prop from "toggledVersion" to "collapsibleVersion".

**Comment by swetharajendra05** [comment]
@mengtzu : added the offset to footer and renamed the props. Please help to review.🙏

**Comment by mengtzu** [comment]
I want to approve it but I'm still worried about the preview being broken - can we pair on that to figure out what's going wrong and fix it?

**Comment by mengtzu** [comment]
:shipit:

File: jobsDB/Footer/Footer.js [hunk 5]
```diff
@@ -1,36 +1,20 @@
...
-            </span>
-          </p>
-        </div>
-      </div>
-    </footer>
-  );
+    <FooterHK
+      language={language}
+      country={country}
+      isExpandedVersion={false}
```
[the remark this answers is not in the document]
Comment (swetharajendra05) [reply jobsDB/Footer/Footer.js]: @chiwongseekasia :  `<Footer/>` refers to the legacy version of the footer which we got rid off. I'm not sure why we need to explicit check `country='HK' ` for `FooterHK` only and not for the legacy `Footer`. Please do let me know if you know. Do agree we can refactor the code further but may be in different PR.

**Pull request merged**